cg/apps/cgstats/crud/find.py

```diff
@@ -23,52 +23,83 @@
 class FindHandler:
     def get_support_parameters_id(self, demux_results: DemuxResults) -> Optional[int]:
         """Fetch the id of the support parameters if post exists"""
-        LOG.debug("Search for support parameters with file %s", demux_results.results_dir)
-        support_parameters_id: Optional[int] = Supportparams.exists(str(demux_results.results_dir))
-        if support_parameters_id:
-            LOG.debug("Found support parameters with id %s", support_parameters_id)
+        LOG.debug(f"Search for support parameters with file {demux_results.results_dir}")
+
+        support_parameters: Supportparams = self.get_support_parameters_by_document_path(
+            document_path=str(demux_results.results_dir)
+        )
+
+        if support_parameters:
+            support_parameters_id = support_parameters.supportparams_id
+            LOG.debug(f"Found support parameters with id {support_parameters_id}")
             return support_parameters_id
+
         LOG.debug("Could not find support parameters")
         return None
 
+    def get_support_parameters_by_document_path(
+        self, document_path: str
+    ) -> Optional[Supportparams]:
+        return Supportparams.query.filter_by(document_path=document_path).first()
+
     def get_datasource_id(self, demux_results: DemuxResults) -> Optional[int]:
         """Fetch the datasource id for a certain run"""
         stats_path = {
             "bcl2fastq": demux_results.conversion_stats_path,
             "dragen": demux_results.demux_stats_path,
         }
-        LOG.debug("Search for datasource with file %s", stats_path[demux_results.bcl_converter])
-        datasource_id: Optional[int] = Datasource.exists(
-            str(stats_path[demux_results.bcl_converter])
-        )
-        if datasource_id:
-            LOG.debug("Found datasource with id %s", datasource_id)
-            return datasource_id
+        document_path: str = str(stats_path[demux_results.bcl_converter])
+        LOG.debug(f"Search for datasource with file {document_path}")
+        datasource: Datasource = self.get_datasource_by_document_path(document_path=document_path)
+
+        if datasource:
+            LOG.debug(f"Found datasource with id {datasource.datasource_id}")
+            return datasource.datasource_id
+
         LOG.debug("Could not find datasource")
         return None
 
+    def get_datasource_by_document_path(self, document_path: str) -> Optional[Datasource]:
+        return Datasource.query.filter_by(document_path=document_path).first()
+
     def get_flowcell_id(self, flowcell_name: str) -> Optional[int]:
```

Comment:
Refactor flowcell to flow_cell as much as possible

Reply:
Renamed some, seems like parameter renaming is not reliable - picks up other instances. Saving the rest for a future PR.